pkg/cluster/streams.go

```diff
@@ -0,0 +1,349 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	acidv1 "github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1"
+	zalandov1 "github.com/zalando/postgres-operator/pkg/apis/zalando.org/v1"
+	"github.com/zalando/postgres-operator/pkg/util"
+	"github.com/zalando/postgres-operator/pkg/util/constants"
+	"github.com/zalando/postgres-operator/pkg/util/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (c *Cluster) createStreams(appId string) {
+	c.setProcessName("creating streams")
+
+	var (
+		fes *zalandov1.FabricEventStream
+		err error
+	)
+
+	msg := "could not create event stream custom resource with applicationId %s: %v"
+
+	fes = c.generateFabricEventStream(appId)
+	if err != nil {
+		c.logger.Warningf(msg, appId, err)
+	}
+	_, err = c.KubeClient.FabricEventStreams(c.Namespace).Create(context.TODO(), fes, metav1.CreateOptions{})
+	if err != nil {
+		c.logger.Warningf(msg, appId, err)
+	}
+}
+
+func (c *Cluster) updateStreams(newEventStreams *zalandov1.FabricEventStream) error {
+	c.setProcessName("updating event streams")
+
+	_, err := c.KubeClient.FabricEventStreams(newEventStreams.Namespace).Update(context.TODO(), newEventStreams, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not update event stream custom resource: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Cluster) deleteStreams() error {
+	c.setProcessName("deleting event streams")
+
+	// check if stream CRD is installed before trying a delete
```

Comment:
that should be a condition for calling `deleteStreams()`

Reply:
I would prefer not to do that in the Delete() function of Cluster, to keep the code there cleaner. I think, it's fine here.